### PR TITLE
BREAKING(fs): remove `Deno.FsWatcher.prototype.return()`

### PR DESCRIPTION
**Pull request by iuioiua**
Towards #22079

### CODE REVIEW

File: tests/unit/fs_events_test.ts [hunk 1]
```diff
@@ -49,7 +49,7 @@ Deno.test(
   { permissions: { read: true, write: true } },
   async function watchFsBasic() {
     const testDir = await makeTempDir();
-    const iter = Deno.watchFs(testDir);
+    using iter = Deno.watchFs(testDir);
```
Comment (iuioiua) [inline tests/unit/fs_events_test.ts:52]: Why was this needed?

File: tests/unit/fs_events_test.ts [hunk 2]
```diff
@@ -74,7 +74,7 @@ Deno.test(
   { permissions: { read: true, write: true } },
   async function watchFsRename() {
     const testDir = await makeTempDir();
-    const watcher = Deno.watchFs(testDir);
+    using watcher = Deno.watchFs(testDir);
```
Comment (iuioiua) [inline tests/unit/fs_events_test.ts:77]: Ditto

File: tests/unit/fs_events_test.ts (hunk 1 above)
Comment (lucacasonato) [reply tests/unit/fs_events_test.ts:52]: So that the watcher gets closed at the end of the test.

Comment (iuioiua) [reply tests/unit/fs_events_test.ts:52]: To clarify, why does removing `Deno.FsWatcher.prototype.return()` cause the `FsWatcher` to not be automatically closed. I guess the method is automatically closed once the iterator is done...

**Comment by iuioiua** [comment]
Tbh, I, and some other contributors, think that we shouldn't be removing this method. And instead, perhaps adding the method to iterables that don't implement it. https://github.com/denoland/deno/issues/10577#issuecomment-1073810140

**Comment by bartlomieju** [comment]
Closing in favor of #25623
